### PR TITLE
Add travis build

### PR DESCRIPTION
**Pull request by snim2**
Use Travis to build Krun and run the example configuration.
Make some Makfiles more portable.
Adds a `krun` user, which is not currently needed, but might be useful when we get rid of `--develop`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
-script: py.test --cov-report term --cov=krun  krun
+script:
+    - py.test --cov-report term --cov=krun  krun
+    - make --version
+    - ldd --version
+    - JAVA_CPPFLAGS='"-I${JAVA_HOME}/include -I${JAVA_HOME}/include/linux"' JAVA_LDFLAGS=-L${JAVA_HOME}/lib ENABLE_JAVA=1 make
+    - cd examples/benchmarks
+    - make
+    - cd ../
+    - PYTHONPATH=../ python ../krun.py --develop example.krun
```
Comment (vext01) [inline .travis.yml]: Nitpicking, but why `../` not `..`?

**Comment by vext01** [comment]
librt (and I think libdl) are linux specific. E.g. on OpenBSD this branch fails to build:

```
wilfred:krun] gmake
cd libkrun && gmake ENABLE_JAVA= JAVAC=javac JAVA_CPPFLAGS= JAVA_CFLAGS= JAVA_LDFLAGS= CC=cc CFLAGS= CPPFLAGS= LDFLAGS=
gmake[1]: Entering directory '/home/krun/krun/libkrun'
cc   -Wall -shared -fPIC  \
         libkruntime.c -o libkruntime.so \
          -lrt -ldl\

/usr/bin/ld: cannot find -lrt
collect2: ld returned 1 exit status
Makefile:13: recipe for target 'libkruntime.so' failed
gmake[1]: *** [libkruntime.so] Error 1
gmake[1]: Leaving directory '/home/krun/krun/libkrun'
Makefile:17: recipe for target 'libkrun' failed
gmake: *** [libkrun] Error 2
```

**Comment by ltratt** [comment]
Travis is, at least for the moment for us, Linux only, so does it matter?

**Comment by vext01** [comment]
The problem is introduced into the Makefile used by all systems.

**Comment by ltratt** [comment]
Ah, I stand corrected!

**Comment by vext01** [comment]
Otherwise this looks good!

File: libkrun/Makefile [hunk 2]
```diff
@@ -1,5 +1,10 @@
 LIBKRUNTIME_CFLAGS =	-Wall -shared -fPIC
-LIBKRUNTIME_LDFLAGS =	-lrt -ldl
+
+ifeq (`uname -s`,Linux)
+	LIBKRUNTIME_LDFLAGS =	-lrt -ldl
+else
+	LIBKRUNTIME_LDFLAGS =	-lrt
```
Comment (vext01) [inline libkrun/Makefile]: No librt on OpenBSD.

**Comment by vext01** [comment]
Works on OpenBSD. Squash away.

**Comment by snim2** [comment]
All ready to merge!

**Comment by snim2** [comment]
Thanks!

**Pull request merged**